### PR TITLE
Change RBAC management policy to Basic by default

### LINKED ISSUE #39: Change RBAC management policy to Basic by default
## Context

Crossplane uses a complex mechanism of handling RBAC rules for users with Claims. See https://github.com/crossplane/crossplane/blob/master/design/design-doc-rbac-manager.md.

We do not have the requirements for such fine grained RBAC permissions yet as set by default. Instead, at this time we've gone for the Basic management policy in APPUiO Cloud and we could consider making this the default in this component.
See also https://github.com/vshn/component-appcat/pull/12

This is a breaking change though.

(side note: It can be a possibility to remove the upgrade job that was engineered for an earlier version of Crossplane, and we're currently ahead)

## ACs

* Add the Helm parameters to the defaults that sets the management policy to "Basic"
* Remove the CRD upgrade job that was introduced in #9 (not needed anymore)
* Documentation
   * Document the minimum version number of Crossplane to v1.7
   * Add a page that describes how to upgrade to the next major version
* Release component as v3.0.0, upgrade it on our APPUiO Cloud zones (commodore defaults)
* Make internal announcement on discuss.vshn.net

## Alternatives

* Each cluster catalog explicitly sets the management policy to Basic (current)

### PR DESCRIPTION
**Pull request by susana-garcia**
## Summary 

* Change RBAC management policy to Basic by default
* Remove the CRD upgrade job (not needed anymore since 1.7)
* This is a breaking change

Fixes #39 

## Checklist

- [X] PR contains a single logical change (to build a better changelog).
- [X] Update the documentation.
- [X] Categorize the PR by setting a good title and adding one of the labels:
      `bug`, `enhancement`, `documentation`, `change`, `breaking`, `dependency`
      as they show up in the changelog.
- [X] Link this PR to related issues or PRs.

### CODE REVIEW

File: tests/defaults-with-controller-configs.yml [hunk 1]
```diff
@@ -75,3 +75,5 @@ parameters:
           cpu: 500m
       metrics:
         enabled: ${crossplane:monitoring:enabled}
+      rbacManager:
+        managementPolicy: Basic
```
Comment (glrf) [inline tests/defaults-with-controller-configs.yml:79]: Nitpick: You don't need this in the tests as it's already specified in the component defaults that will be merged into this testcase

File: README.md [hunk 2]
```diff
@@ -17,6 +17,14 @@ Run the `make docs-serve` command in the root of the project, and then browse to
 
 After writing the documentation, please use the `make docs-vale` command and correct any warnings raised by the tool.
 
+## Local installation for testing purposes
+
+`make install` allows you to install the operator in a local (kind) cluster.
+
+This target installs the Prometheus Operator, according to the instructions explained [here](https://github.com/prometheus-operator/prometheus-operator#quickstart)
+
+Note: the namespace in the ClusterRoleBinding needs to be updated as we're deploying in a namespace other than the default namespace, that's why we have `kind/prometheus-operator-cluster-role-binding.yaml`
```
Comment (glrf) [inline README.md:26]: Nitpick: It's a bit weird that you primarily talk about the Prometheus Operator. At first I was confused what component I'm reviewing again :sweat_smile:

**Review by glrf — APPROVED** [review]
LGTM overall.

However, this is technically a breaking change. Not sure if we really want to release v3 for this though as I don't know if anyone is actually using the `All` RBAC management policy..

File: tests/defaults-with-controller-configs.yml (hunk 1 above)
Comment (susana-garcia) [reply tests/defaults-with-controller-configs.yml]: you're absolutely right, I deleted it, thanks

File: README.md (hunk 2 above)
Comment (susana-garcia) [reply README.md:26]: yes, that's true, now that you said it, it's a bit confusing. I'll update the explanation

**Comment by susana-garcia** [comment]
> LGTM overall.
> 
> However, this is technically a breaking change. Not sure if we really want to release v3 for this though as I don't know if anyone is actually using the `All` RBAC management policy..

Yes, that's a good point. What do you think @Kidswiss ?

File: README.md [hunk 3]
```diff
@@ -17,6 +17,14 @@ Run the `make docs-serve` command in the root of the project, and then browse to
 
 After writing the documentation, please use the `make docs-vale` command and correct any warnings raised by the tool.
 
+## Local installation for testing purposes
+
+`make install` allows you to install the operator in a local (kind) cluster.
+
+This targe installs the component Crossplane in a local cluster. This component uses the Prometheus Operator and to install the last one, we use this [guide](https://github.com/prometheus-operator/prometheus-operator#quickstart)
```
Comment (zugao) [inline README.md:24]:
```suggestion
The target installs the component Crossplane in a local cluster. This component uses [Prometheus Operator](https://github.com/prometheus-operator/prometheus-operator#quickstart).
```

**Review by zugao — APPROVED** [review]
LGTM, some typos

**Comment by Kidswiss** [comment]
> Yes, that's a good point. What do you think @Kidswiss ?

Not quite sure what the question is here... If we should keep 2.x or go to v3?

**Comment by susana-garcia** [comment]
> > Yes, that's a good point. What do you think @Kidswiss ?
> 
> Not quite sure what the question is here... If we should keep 2.x or go to v3?

Yes, sorry, that's the question. This PR solves issue #39 and it's a breaking change. There is suggested to create a new version 3, but as no one seems to be using this component yet, doesn't really matter. But for consistency, it might be a good idea to create a new version v3.

**Comment by Kidswiss** [comment]
> > > Yes, that's a good point. What do you think @Kidswiss ?
> > 
> > 
> > Not quite sure what the question is here... If we should keep 2.x or go to v3?
> 
> Yes, sorry, that's the question. This PR solves issue #39 and it's a breaking change. There is suggested to create a new version 3, but as no one seems to be using this component yet, doesn't really matter. But for consistency, it might be a good idea to create a new version v3.

Ah I see... I'd still go for consistency, though.

**Comment by susana-garcia** [comment]
> > > > Yes, that's a good point. What do you think @Kidswiss ?
> > > 
> > > 
> > > Not quite sure what the question is here... If we should keep 2.x or go to v3?
> > 
> > 
> > Yes, sorry, that's the question. This PR solves issue #39 and it's a breaking change. There is suggested to create a new version 3, but as no one seems to be using this RBAC yet, doesn't really matter. But for consistency, it might be a good idea to create a new version v3.
> 
> Ah I see... I'd still go for consistency, though.

ok, thank you 👍

**Pull request merged**